### PR TITLE
feat:[CI-10505]:adds oidc support for GAR buildx

### PR DESCRIPTION
**Pull request by eoinmcafee00**
Adds GAR OIDC support for buildx in hosted.

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55821)

### CODE REVIEW

**Comment by eoinmcafee00** [comment]
trigger uts

**Review by harshjain12 — APPROVED** [review]

File: 933-ci-commons/src/main/java/io/harness/vm/VmExecuteStepUtils.java [hunk 1]
```diff
@@ -162,8 +162,18 @@ private List<String> setRunConfig(VmRunStep runStep, ConfigBuilder configBuilder
...
-    }
+      // Map OIDC config to plugin env variables
+      List<EnvVariableEnum> keyList = Arrays.asList(PLUGIN_POOL_ID, PLUGIN_PROJECT_NUMBER, PLUGIN_PROVIDER_ID,
+          PLUGIN_SERVICE_ACCOUNT_EMAIL, PLUGIN_OIDC_TOKEN_ID);
+
+      Map<EnvVariableEnum, String> envToSecretsMap = runStep.getConnector().getEnvToSecretsMap();
 
+      for (EnvVariableEnum key : keyList) {
+        if (envToSecretsMap.containsKey(key)) {
+          envs.put(key.name(), envToSecretsMap.get(key));
```
Comment (harshjain12) [inline 933-ci-commons/src/main/java/io/harness/vm/VmExecuteStepUtils.java:173]: can you check if env is not empty

Comment (eoinmcafee00) [reply 933-ci-commons/src/main/java/io/harness/vm/VmExecuteStepUtils.java:173]: What would the benefit be? Not seeing the issue?

Comment (harshjain12) [reply 933-ci-commons/src/main/java/io/harness/vm/VmExecuteStepUtils.java:173]: It can throw null pointer exception if envs is null

**Review by harshjain12 — APPROVED** [review]

**Pull request merged**